### PR TITLE
grey out on timeslot and make class rating on class images

### PR DESCRIPTION
**Pull request by kimgayeon**
grey out on timeslot if it is reserved
make class rating on class images

### CODE REVIEW

File: Android/app/src/main/java/uncommon/common/activity/ReservationActivity.java [hunk 1]
```diff
@@ -164,7 +205,8 @@ public void onClick(View view) {
                 AlertDialog.Builder builder = new AlertDialog.Builder(ReservationActivity.this);
                 //TODO (woongjin) need to refactor this block
                 //now its spaghetti code
-                if(selectedTime != null && ticketCount != 0){
+                if(selectedTime != null && ticketCount != 0
```
Comment (Ricci0708) [inline Android/app/src/main/java/uncommon/common/activity/ReservationActivity.java]: For the readability, using De Morgan's laws pop out the NOT operation outside
if(!(selectedTime == null || ticketCount == 0 || timeslot.get(selectedTimeSlotIdx).getIsBooked()))

File: Android/app/src/main/java/uncommon/common/activity/ReservationActivity.java [hunk 2]
```diff
@@ -93,14 +98,29 @@ protected void onCreate(Bundle savedInstanceState) {
         TextView dateView = (TextView) findViewById(R.id.dateView);
         dateView.setText(selectedDate);
 
+        //(TODO) gayeon : change the date
```
Comment (ethkim) [inline Android/app/src/main/java/uncommon/common/activity/ReservationActivity.java]: Change the date to what? Describe it fully in the comment.

File: Android/app/src/main/java/uncommon/common/activity/ReservationActivity.java [hunk 3]
```diff
@@ -109,7 +129,28 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        // grey out on time slot
+        final ArrayAdapter<String> timeslotAdapter = new ArrayAdapter<String>
+                (this, android.R.layout.simple_list_item_1, timeList){
+            @Override
+            public View getView(int position, View convertView, ViewGroup parent){
+                // Get the current item from ListView
+                View view = super.getView(position, convertView, parent);
+
+                boolean isBooked = timeslot.get(position).getIsBooked();
+                if(isBooked) {
```
Comment (ethkim) [inline Android/app/src/main/java/uncommon/common/activity/ReservationActivity.java]: Space before (

if (isBooked) {

File: Android/app/src/main/java/uncommon/common/activity/ReservationActivity.java [hunk 4]
```diff
@@ -109,7 +129,28 @@ protected void onCreate(Bundle savedInstanceState) {
...
         }
 
-        timeListView.setAdapter(adapter);
+        // grey out on time slot
+        final ArrayAdapter<String> timeslotAdapter = new ArrayAdapter<String>
+                (this, android.R.layout.simple_list_item_1, timeList){
+            @Override
+            public View getView(int position, View convertView, ViewGroup parent){
+                // Get the current item from ListView
+                View view = super.getView(position, convertView, parent);
```
Comment (ethkim) [inline Android/app/src/main/java/uncommon/common/activity/ReservationActivity.java]: I think (I may be wrong) you can cast view to TextView, so you don't need to the casting below.

File: Android/app/src/main/java/uncommon/common/activity/ReservationActivity.java (hunk 1 above)
Comment (ethkim) [reply Android/app/src/main/java/uncommon/common/activity/ReservationActivity.java]: I would actually just quit this function entirely if (selectedTime == null || ticketCount == 0 || ...)

Then continue with the graying out operation below - so you do the error handling before, and run the normal logic in the main body.

File: Android/app/src/main/java/uncommon/common/utils/ChangeDate.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import android.content.Context;
+import android.view.View;
+import android.widget.DatePicker;
+import android.widget.TextView;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Locale;
+
+public class ChangeDate implements View.OnFocusChangeListener, DatePickerDialog.OnDateSetListener {
```
Comment (ethkim) [inline Android/app/src/main/java/uncommon/common/utils/ChangeDate.java]: Switch DatePickerDialog and View.

File: Android/app/src/main/java/uncommon/common/activity/ReservationActivity.java (hunk 1 above)
Comment (Ricci0708) [reply Android/app/src/main/java/uncommon/common/activity/ReservationActivity.java]: OK I will handle this problem

**Pull request merged**